File: python/recursion/sum_array.py
from typing import List
def sumArray(arr: List[int])->int:
    '''
    return the sum of the given integer list using recursive call 
    input: List[int]
    output: int
    '''
    if len(arr) == 1:
        return arr[0]    
    return arr[0] + sumArray(arr[1:])
    
arr = [1, 2, 3, 4]
print(sumArray(arr))

File: mycodes/arrays/sum_array.cpp
#include<bits/stdc++.h>
using namespace std;
void input(int arr[], int n){
    for(int i = 0; i < n; i++){
        cin >> arr[i];
    }
}
int sum(int arr[], int n){
    int sum = 0;
    for(int i = 0; i < n; i++){
        sum += arr[i];
    }
    return sum;

}

int main(){
    int n;
    cin >> n;
    int arr[1000];
    input(arr, n);
    cout << "Sum of the array: " << sum(arr, n) << endl;
    return 0;
}